include/reactor-uc/network_channel.h

```diff
@@ -31,14 +31,26 @@ typedef enum {
 typedef enum {
   NETWORK_CHANNEL_TYPE_TCP_IP,
   NETWORK_CHANNEL_TYPE_COAP_UDP_IP,
+  NETWORK_CHANNEL_TYPE_UART
 } NetworkChannelType;
 
+typedef enum {
+  NETWORK_CHANNEL_MODE_ASYNC,
+  NETWORK_CHANNEL_MODE_POLL,
+} NetworkChannelMode;
```

Comment:
Maybe call it `NetworkChannelExecutionMode` to be more clear what it is for

Reply:
Hmm, I think mode is OK for now, if we need to add more properties then we should consider being more specific